src/main/java/org/popcraft/chunky/watchdog/WatchdogManager.java

```diff
@@ -0,0 +1,80 @@
+package org.popcraft.chunky.watchdog;
+
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.popcraft.chunky.Chunky;
+import org.popcraft.chunky.GenerationTask;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Level;
+
+public class WatchdogManager implements Runnable {
+
+    private Chunky chunky;
+    private boolean previousState = false;
+    private AbstractGenerationWatchdog previousUnmet;
+    private List<AbstractGenerationWatchdog> watchdogs = new ArrayList<>();
+
+    public WatchdogManager(Chunky chunky) {
+        this.chunky = chunky;
+    }
+
+    @Override
+    public void run() {
+        Optional<AbstractGenerationWatchdog> checkResult = allCanRunOrUnmetWatchdog();
+        boolean canRun = !checkResult.isPresent();
+        if(canRun != previousState) {
+            previousState = canRun;
+            if(canRun) {
+                if(previousUnmet != null) {
+                    Bukkit.getLogger().log(Level.INFO, chunky.message(
+                            previousUnmet.startReasonKey(), chunky.message("prefix")));
+                }
+                start();
+            } else {
+                previousUnmet = checkResult.get();
+                Bukkit.getLogger().log(Level.INFO, chunky.message(
+                        checkResult.get().stopReasonKey(), chunky.message("prefix")));
+                stop();
+            }
+        }
+    }
+
+    private Optional<AbstractGenerationWatchdog> allCanRunOrUnmetWatchdog() {
+        AbstractGenerationWatchdog unmet = null;
+        for (AbstractGenerationWatchdog watchdog : watchdogs) {
+            if(!watchdog.getCanRun()) {
+                unmet = watchdog;
+            }
+        }
+        return Optional.ofNullable(unmet);
+    }
+
+    private void start() {
+        final List<GenerationTask> loadTasks = chunky.getConfigStorage().loadTasks();

```

Comment:
This would load tasks from the config including those that the user is not currently running